### PR TITLE
Update examples with `@node` directive

### PR DESCRIPTION
**Pull request by MacondoExpress**
Add `@node` in the examples and update the migration guide from 4.x to 5.x to be from 5.x to 6.x.

### CODE REVIEW

**Comment by MacondoExpress** [comment]
@rsill-neo4j  we opened a PR similar before, but now we introduce this warnings in v6 rather than on v5.

File: modules/ROOT/pages/directives/database-mapping.adoc [hunk 1]
```diff
@@ -8,84 +8,26 @@ This page describes how to use directives for database mapping.
...
     title: String
 }
 ----
 
-With the `@node` directive, you have more control over this mapping, and you can specify the configuration of a GraphQL object type which represents a Neo4j node.
-It can be appended with the following optional parameters:
+[NOTE]
+====
+In the current version, it's not required to specify every GraphQL type representing a Neo4j node with the `@node` directive.
+In the future, types without the `@node` directive will no longer treated as Neo4j nodes.
```
Comment (rsill-neo4j) [inline modules/ROOT/pages/directives/database-mapping.adoc]:
```suggestion
In the future, types without the `@node` directive will no longer be treated as Neo4j nodes.
```

File: modules/ROOT/pages/directives/database-mapping.adoc [hunk 2]
```diff
@@ -8,84 +8,26 @@ This page describes how to use directives for database mapping.
...
+type Movie @node {
     title: String
 }
 ----
 
-With the `@node` directive, you have more control over this mapping, and you can specify the configuration of a GraphQL object type which represents a Neo4j node.
-It can be appended with the following optional parameters:
+[NOTE]
+====
+In the current version, it's not required to specify every GraphQL type representing a Neo4j node with the `@node` directive.
```
Comment (rsill-neo4j) [inline modules/ROOT/pages/directives/database-mapping.adoc]:
```suggestion
In version 6.x, it's not required to specify every GraphQL type representing a Neo4j node with the `@node` directive.
```

File: modules/ROOT/pages/directives/database-mapping.adoc [hunk 3]
```diff
@@ -216,6 +158,67 @@ await startStandaloneServer(server, {
 });
 ----
 
+== `@relationship`
+
+Relationships are represented by marking particular fields with a directive -- in this case, `@relationship`. 
+It defines the relationship type in the database, as well as which direction that relationship goes in.
```
Comment (rsill-neo4j) [inline modules/ROOT/pages/directives/database-mapping.adoc]:
```suggestion
It defines the relationship type in the database, as well as it's direction.
```

File: modules/ROOT/pages/directives/database-mapping.adoc [hunk 4]
```diff
@@ -216,6 +158,67 @@ await startStandaloneServer(server, {
...
+
+To add a second node type, "Actor", and connect the two together, you should do the following:
```
Comment (rsill-neo4j) [inline modules/ROOT/pages/directives/database-mapping.adoc]:
```suggestion
The following type definitions add a second node type, `Actor`, and connect `Movie` and `Actor` via the `ACTED_IN` relationship:
```

File: modules/ROOT/pages/directives/database-mapping.adoc [hunk 5]
```diff
@@ -216,6 +158,67 @@ await startStandaloneServer(server, {
...
+    actors: [Actor!]! @relationship(type: "ACTED_IN", direction: IN)
+}
+
+type Actor @node {
+    name: String
+    movies: [Movie!]! @relationship(type: "ACTED_IN", direction: OUT)
+}
+----
+
+Note that, in this case, there is a directive on each "end" of the relationship, but it is not essential.
```
Comment (rsill-neo4j) [inline modules/ROOT/pages/directives/database-mapping.adoc:181]:
```suggestion
[NOTE]
====
The `@relationship` directive is used twice, once on each end of the relationship.
This is the standard way of modeling a relationship with the GraphQL library.
However, it is not a requirement of the type definitions themselves and relationships can deliberately be underspecified, for example to limit access through the API layer.
====
```

Please double-check that it's correct(ly phrased) what i stated

File: modules/ROOT/pages/migration/index.adoc [hunk 6]
```diff
@@ -18,12 +18,16 @@ npm update @neo4j/graphql
...
 == Breaking changes
 
-Here is a list of all the breaking changes from version 4.0.0 to 5.0.0.
+Here is a list of all the breaking changes from version 5.0.0 to 6.0.0.
 
-=== `@relationshipProperties`
 
-The directive `@relationshipProperties` should now be used on types rather than interfaces.
-For example:
+== Deprecations and Warnings
```
Comment (rsill-neo4j) [inline modules/ROOT/pages/migration/index.adoc]:
```suggestion
== Deprecations and warnings
```

File: modules/ROOT/pages/migration/index.adoc [hunk 7]
```diff
@@ -18,12 +18,16 @@ npm update @neo4j/graphql
...
+
+=== `@node` will be required to be explicitly defined
```
Comment (rsill-neo4j) [inline modules/ROOT/pages/migration/index.adoc]:
```suggestion
=== `@node` will have to be explicitly defined
```

File: modules/ROOT/pages/migration/index.adoc [hunk 8]
```diff
@@ -18,12 +18,16 @@ npm update @neo4j/graphql
...
+
+In the future, types without the `@node` directive will no longer be treated as Neo4j nodes.
+In version 6.0.0, it's not required to specify every GraphQL type representing a Neo4j node with the `@node` directive, however it's recommended to do so and a warning is raised if not found.
```
Comment (rsill-neo4j) [inline modules/ROOT/pages/migration/index.adoc]:
```suggestion
In version 6.0.0, it's not required to specify every GraphQL type representing a Neo4j node with the `@node` directive, however it's recommended to do so and a warning is raised if you don't.
```

File: modules/ROOT/pages/migration/index.adoc [hunk 9]
```diff
@@ -18,12 +18,16 @@ npm update @neo4j/graphql
...
+There are genuine cases in which a `@node` should not be used e.g., for type returned by @cypher fields which are not Neo4j nodes.
```
Comment (rsill-neo4j) [inline modules/ROOT/pages/migration/index.adoc]:
```suggestion
There are genuine cases in which a `@node` should not be used, e.g., for types returned by @cypher fields which are not Neo4j nodes.
```

File: modules/ROOT/pages/queries-aggregations/filtering.adoc [hunk 10]
```diff
@@ -103,7 +103,108 @@ query {
...
+        }
+    }
+};
+
+const neoSchema = new Neo4jGraphQL({ features, typeDefs, driver });
+----
+
+== RegEx matching
+
+The filter `_MATCHES` is also available for comparison of `String` and `ID` types.
```
Comment (rsill-neo4j) [inline modules/ROOT/pages/queries-aggregations/filtering.adoc]: this file/section has been updated (recently, 10th of october) in 6.x:
https://github.com/neo4j/docs-graphql/blob/6.x/modules/ROOT/pages/queries-aggregations/filtering.adoc#regex-matching

not sure why the changes didn't make it here :thinking:

File: modules/ROOT/pages/directives/database-mapping.adoc (hunk 5 above)
Comment (MacondoExpress) [reply modules/ROOT/pages/directives/database-mapping.adoc:181]: Let's keep this outside this PR as it is as it is on 6.x, the only reason why it is marked as a changed is because I moved the `@node` description at the top of the page to remark the new importance.

File: modules/ROOT/pages/queries-aggregations/filtering.adoc (hunk 10 above)
Comment (MacondoExpress) [reply modules/ROOT/pages/queries-aggregations/filtering.adoc]: Good catch! let me check it!

File: modules/ROOT/pages/directives/database-mapping.adoc [hunk 11]
```diff
@@ -216,6 +158,67 @@ await startStandaloneServer(server, {
 });
 ----
 
+== `@relationship`
+
+Relationships are represented by marking particular fields with a directive -- in this case, `@relationship`. 
+It defines the relationship type in the database, as well as it's direction.
```
Comment (rsill-neo4j) [inline modules/ROOT/pages/directives/database-mapping.adoc]:
```suggestion
It defines the relationship type in the database, as well as its direction.
```

my own typo ;)

**Review by rsill-neo4j — APPROVED** [review]
